coderd/notifications/reports/generator.go

```diff
@@ -49,7 +49,7 @@ func NewReportGenerator(ctx context.Context, logger slog.Logger, db database.Sto
 				return nil
 			}
 
-			err = reportFailedWorkspaceBuilds(ctx, logger, db, enqueuer, clk)
+			err = reportFailedWorkspaceBuilds(ctx, logger, tx, enqueuer, clk)
```

Comment:
Passing the main `db` object through caused a deadlock; we need to pass the transaction-scoped db object instead.